props/prop_base.h

```diff
@@ -53,6 +53,51 @@ class SavePresetStateFile : public ConfigFile {
 #endif    
 };
 
+struct SoundToPlay {
+  const char* filename_;
+  Effect* effect_;
+  int selection_;
+
+  SoundToPlay() :filename_(nullptr), effect_(nullptr) {}
+  SoundToPlay(const char* file) : filename_(file){  }
+  SoundToPlay(Effect* effect, int selection = -1) : filename_(nullptr), effect_(effect), selection_(selection) {}
+  bool Play(BufferedWavPlayer* player) {
+     if (filename_)   return player->PlayInCurrentDir(filename_);
+     effect_->Select(selection_);
+     player->PlayOnce(effect_);
+     return true;
+   }
+   bool isSet() {
+      return filename_ != nullptr || effect_ != nullptr;
+   }
+};
+
+template<int QueueLength>
+class SoundQueue {
+public:
+   bool Play(SoundToPlay p) {
+     if (sounds_ < QueueLength) {
+       queue_[sounds_++] = p;
+       return true;
+     }
+     return false;
+   }
+   // Called from Loop()
+ void Poll(RefPtr<BufferedWavPlayer>& player) {
```

Comment:
Looks like it needs to be indented two more steps?

Reply:
fixed indent